test/framework/ownerreference_helpers.go

```diff
@@ -184,16 +189,16 @@ var (
 var ExpOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error{
 	clusterResourceSetKind: func(owners []metav1.OwnerReference) error {
 		// ClusterResourcesSet doesn't have ownerReferences (it is a clusterctl move-hierarchy root).
-		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{})
+		return HasExactOwners(owners)
 	},
 	// ClusterResourcesSetBinding has ClusterResourceSet set as owners on creation.
 	clusterResourceSetBindingKind: func(owners []metav1.OwnerReference) error {
-		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterResourceSetGVK})
+		return HasOneOfExactOwners(owners, []simpleOwnerRef{clusterResourceSetOwner}, []simpleOwnerRef{clusterResourceSetOwner, clusterResourceSetOwner})
```

Comment:
ClusterResourceSetBinding will have as many owners as ClusterResourceSets it managed for a single cluster. I added the option of 2 here just to save me a rewrite in CAPV - can revert if you'd prefer.  

I think this is one of the ownerReference that we should reconsider.

Reply:
Ah I see because doesn't matter how many CRS we have we will only have one ClusterResourceSetBinding per cluster, right?

(not really familiar with this stuff)

Reply:
Yeah - each Cluster will have one CRS binding, but each binding may represent multiple ClusterResourceSets